src/preprocess.js

```diff
@@ -56,6 +58,210 @@ ${diagramText}
   return JSON.stringify(diagramObject)
 }
 
+// @ts-check
+/**
+ * @param {string} diagramText
+ * @param {any} context
+ * @returns {string}
+ */
+module.exports.preprocessPlantUML = function (diagramText, context) {
+  let vfs = context.vfs
+  if (typeof vfs === 'undefined' || typeof vfs.read !== 'function' || typeof vfs.exists !== 'function') {
+    vfs = require('./node-fs.js')
+  }
+  const includeOnce = []
+  const includeStack = []
+  return preprocessPlantUmlIncludes(diagramText, '.', includeOnce, includeStack, vfs)
```

Comment:
I'm not sure how relative paths are resolved on PlantUML. Is the relative path resolved relatively to the PlantUML file itself?

```
/path/to/dir/a.puml
/path/to/b.puml
/path/to/c.puml
/path/to/dir/subdir/d.puml
/path/to/e.puml
```

`/path/to/dir/a.puml` includes `../b.puml`
`/path/to/b.puml` includes `./c.puml`
`c.puml` includes `./dir/subdir/d.puml`
`./dir/subdir/d.puml` includes `../../e.puml`

Is that how it's supposed to work?

Reply:
Yes, that how it's supposed to work - PlantUML resolves the included files relative to the file that includes them.